File: app/src/main/java/com/wys/baseframedemo/utils/MyLog.java
package com.wys.baseframedemo.utils;

import android.util.Log;

import com.wys.baseframedemo.common.MConfiger;

/***
 * 日志类
 */
public class MyLog {
	
	public static final String Billy = "Billywen";

	public static final boolean isReportDebugable(){
		return MConfiger.isReportDebug;
	}
	
	/****
	 * Json解析相关开关
	 * @return
	 */
	public static final boolean isJsonDebugable(){
		return MConfiger.isJSONDebug;
	}
	
	/****
	 * 是否debug调试模式
	 * @return
	 */
	public static final boolean isDebugable(){
		return MConfiger.isDebug;
	}

	/****
	 * Img调试开关
	 * @return
	 */
	public static final boolean isImgDebugable(){
		return MConfiger.isImgDebug;
	}

	/***
	 * 下拉刷新调试相关开关
	 * @return
	 */
	public static final boolean isNLRefreshDebugable(){
		return MConfiger.isNLPullDebug;
	}
	
	
	public static final void debug(String TAG, String msg){
		if(!MConfiger.isDebug){
			return;
		}
		Log.d(TAG,msg);
	}
	
	public static final void error(String TAG, String msg, Throwable throwable){
		if(!MConfiger.isDebug){
			return;
		}
		Log.e(TAG, msg, throwable);
	}
	
	public static final void error(String TAG, String msg){
		if(!MConfiger.isDebug){
			return;
		}
		Log.e(TAG, msg);
	}

	public static final void error(String TAG, Throwable throwable){
		if(!MConfiger.isDebug){
			return;
		}
		Log.e(TAG, "", throwable);
	}
}

File: app/src/main/java/com/beta/MoneyballMaster/utils/MyLog.kt
package com.beta.MoneyballMaster.utils

import android.util.Log

import com.beta.MoneyballMaster.common.MConfiger

/***
 * 日志类
 */
object MyLog {


    val isReportDebugable: Boolean
        get() = MConfiger.isReportDebug

    /****
     * Json解析相关开关
     * @return
     */
    val isJsonDebugable: Boolean
        get() = MConfiger.isJSONDebug

    /****
     * 是否debug调试模式
     * @return
     */
    val isDebugable: Boolean
        get() = MConfiger.isDebug

    /****
     * Img调试开关
     * @return
     */
    val isImgDebugable: Boolean
        get() = MConfiger.isImgDebug

    /***
     * 下拉刷新调试相关开关
     * @return
     */
    val isNLRefreshDebugable: Boolean
        get() = MConfiger.isNLPullDebug


    fun debug(TAG: String, msg: String) {
        if (!MConfiger.isDebug) {
            return
        }
        Log.d(TAG, msg)
    }

    fun error(TAG: String, msg: String, throwable: Throwable) {
        if (!MConfiger.isDebug) {
            return
        }
        Log.e(TAG, msg, throwable)
    }

    fun error(TAG: String, msg: String) {
        if (!MConfiger.isDebug) {
            return
        }
        Log.e(TAG, msg)
    }

    fun error(TAG: String, throwable: Throwable) {
        if (!MConfiger.isDebug) {
            return
        }
        Log.e(TAG, "", throwable)
    }
}
